src/Hashtable.cpp

```diff
@@ -0,0 +1,155 @@
+#include "Hashtable.h"
+#include <iostream>
+#include "../ShareContainer.h"
+#include "../Share.h"
+#include "../HashUtilities.h"
+#include <string>
+
+Hashtable::Hashtable(int numberOfShares)
+{
+    this->tableSize = 2 * numberOfShares;
+    while(!isPrime(tableSize))
+        tableSize++;                // table size is the a prime bigger than 2 * maximum number of shares
+                                    // ensures a load factor of less than 50%
+
+    ContainerTable = new ShareContainer*[tableSize];  // creates array of pointers
+    for(int i = 0; i < tableSize; i++)
+        this->ContainerTable[i] = NULL;
+}
+
+
+Hashtable::~Hashtable()
+{
+    delete[] ContainerTable;
+}
+
+
+
+void Hashtable::Insert(ShareContainer* ShareContainerAdress, std::string key)
+{
+    Share* ShareAdress = ShareContainerAdress->GetValue();
+    int HashCode = HashUtilities::HashString(key);
+    int index = CalculateIndex(HashCode);
+    int originalIndex = index;
+    int i = 1;
+
+    while(!isEmpty(index))
+    {
+        if(ShareAdress->Equals(ContainerTable[index]->GetValue()))
+        {
+            std::cout << "error, share already exists";
+            return;
+        }
+        index = (originalIndex + (i*i)) % tableSize;
+        i++;
+    }
+
+    ContainerTable[index] = ShareContainerAdress;
+}
+
+
+
+
+int Hashtable::Search(std::string key)
+{
```

Comment:
Du hast da eine Methode zuviel atm macht das nichts anderes als "return Find(key);" sowas kann sinn machen wenn du in der übergeordneten Methode noch irgendwelche Vereinfachungen durchführst.